### DISCUSSION TITLE
AppIcon Badge

### DISCUSSION THREAD

**Discussion by KSemenenko** [CommunityToolkit/Maui#734] [New Feature Discussions]
<img width="220" alt="Screenshot 2022-11-07 at 13 46 05" src="https://user-images.githubusercontent.com/4385716/200313918-3fee3f29-dbbf-4665-8eb1-461bdb0e0486.png">
Let's add badge management to the app icon, shall we?

https://developer.apple.com/documentation/uikit/uiapplication/1622918-applicationiconbadgenumber
https://stackoverflow.com/questions/2905542/is-there-a-way-to-add-a-badge-to-an-application-icon-in-android?rq=1
https://github.com/leolin310148/ShortcutBadger

**Comment by VladislavAntonyuk** [comment]
https://vladislavantonyuk.azurewebsites.net/articles/AppIcon-Badge-with-.NET-MAUI

https://github.com/VladislavAntonyuk/MauiSamples/tree/main/MauiBadge

**Reply by KSemenenko** [reply]
it's so cool, you just do incredible things!

**Reply by JoonghyunCho** [reply]
Very impressive that you also have done Tizen 😆

**Reply by VladislavAntonyuk** [reply]
I just published an article, and I feel that Tizen fans managed to exceed the free limit on Azure app service 😄 I hope the website will be up and running in an hour

**Reply by Ghostbird** [reply]
Hmm… I thought this was never going to work cross platform, because Android forbids badges in app icons. However, as I looked into that before posting this, I realised that those are [Google Play Guidelines](https://developer.android.com/distribute/google-play/resources/icon-design-specifications), and therefore do not necessarily apply to changes to an icon after the app's been installed.

Does anyone know the Android rules for this use case?

**Reply by VladislavAntonyuk** [reply]
@Ghostbird are you talking about this image?
<img width="449" alt="image" src="https://user-images.githubusercontent.com/33021114/208327164-0552ab76-a0ca-4dce-a2cc-06fcec7f3b4f.png">

Here it is a part of an icon (the user drew a png and set it as an icon). This feature is about a badge.

**Reply by Ghostbird** [reply]
I get that. The point is these guidelines by Google forbid app icons including such symbols. However I think it only applies to Google Play store icons. Not a post-install app icon. The difference is that the former could be used to misleadingly advertise your app, the second is user facing after they've already installed your app.

I just wondered if anyone knew for certain.

Context: I think this is already known by all discussion participants, but for anyone who stumbles upon this and only know the iOS side of things: Android doesn't _have_ a badge mechanism _except_ redrawing the icon PNG with an included badge graphic, and setting that as the app's main icon.

**Reply by Ghostbird** [reply]
A bit of a late follow-up, the implementation only works on recent Android versions that have a badge mechanism. I had thought that the _intended_ implementation was to rewrite the app icon internally to bake the badge into the app icon PNG, which would be a violation of the guidelines.

Clearly there's no conflict with the Google Play Guidelines in the chosen implementation.

**Comment by TheCodeTraveler** [comment]
Implemented in https://github.com/CommunityToolkit/Maui/pull/1126
